### DISCUSSION TITLE
常见问题

### DISCUSSION THREAD

**Discussion by VermiIIi0n** [VermiIIi0n/fuckZHS#25] [General]
### 在这里汇总一些经常被问到的问题

* 登入失败
  * 检查您的账号密码是否有误
  * 先用浏览器登入一次看看, 可能您的 IP 地址被认为是异地了, 需要短信验证
  * 解决 SSL 证书错误, 见下文
* Captcha Required
  * 浏览器登入后手动过一次验证
  * 降低单次学习的时间
  * 本脚本会随机暂停一分钟以缓解这个问题, 您可以在源码中增加概率
* 请求响应: -1 "无法处理请求"
  * 某些请求被认为内容不合法了, 因为我测试例很少, 可能有些特例覆盖不全, 请把错误日志贴上, 开个 issue, 我尽力解决
* 语法错误
  * 喂, 伙计, _Python_ 版本对了没?
* 终端二维码乱码 #22 
* SSL 找不到证书, 某些方式安装在 _MacOS_ 的 _Python_ 有这个问题, 详见 #28 
* SSL证书错误, _Windows_ 上开着 _Clash_ 之类的代理可能会报证书错误
  * 直接关闭系统代理
  * 将 _Clash_ 改为 TUN 模式, 且关闭系统代理, 相当于改成透明代理, 不影响上网 ***(推荐)***
  * 配置文件中编辑 `"proxies":{"https":""}`
* _Windows_ 二维码过期 (_Windows_ 奇怪的 bug, 就算独立线程运行显示二维码的函数也会导致程序卡死无法建立 WS 连接, 进而显示为过期)
  * 将 `show_in_terminal` 设为 `true`, 以把二维码显示到终端(还得确保用 Powershell 运行)
* _Windows_ 句柄错误:
  * ~_PyCharm_ 直接运行会这样, 请更换终端~ _FIXED_
* 其他问题
  * 换用 *unix 系统保平安
* 还是无法解决?
  * 建议先开启 DEBUG 模式, 自行查看报错信息, 如果确实是我的锅就劳烦开个 issue
  * 解决提出问题的人 ***(推荐)***  
  ***JUST DO IT!***  
    ![just_do_it](https://user-images.githubusercontent.com/104612478/194684615-24005a87-75dd-4f48-a7cf-c748dc9028ba.jpg)

**Comment by zxz1314258** [comment]
为什么我关闭了二维码为false，输入账号和密码在上面，还是返回给我一个二维码 登录呢

**Comment by zxz1314258** [comment]
还有个问题，我生成的二维码扫了无效果，测试了很多遍，扫了没用

**Reply by Cinnaki** [reply]
我也一样有这个问题，请问你现在解决了吗

**Reply by RavenZal** [reply]
> 我也一样有这个问题，请问你现在解决了吗

先在浏览器上登录一遍

**Reply by Cinnaki** [reply]
> > 我也一样有这个问题，请问你现在解决了吗
> 
> 先在浏览器上登录一遍

浏览器登录过以后还是不行，不过还是谢谢了

**Reply by MONIKA-917** [reply]
微信扫码似乎不行，我用知到app 就能成功

**Reply by jim-orange** [reply]
> 微信扫码似乎不行，我用知到app 就能成功

是得用知到里的扫

**Reply by baida9733** [reply]
用知道里面扫了也是登不上显示401

**Reply by Melgiades** [reply]
昨天用知到可以扫上，今天试了很多遍没反应如何解决

**Comment by CatzTimes** [comment]
##Failed: Expecting value: line 1 column 1 (char 0)
您好 这个是什么问题

**Reply by VermiIIi0n** [reply]
信息不够呢，请按 issue 模板开一个，只看这句话我也没法判断是什么问题。

**Comment by zaitianshangxieshi** [comment]
ImportError: cannot import name 'Fucker' from 'fucker'什么原因捏

**Reply by VermiIIi0n** [reply]
这个不太了解啊, 你有更改过 fucker.py 的内容或者是安装了同名的包吗?

**Reply by 10061hao** [reply]
Captcha required
是为什么啊:(

**Reply by usernameisnotavaliablee** [reply]
> Captcha required 是为什么啊:(

是不是速率太高了 要求验证码人机验证

**Comment by nightzora** [comment]
纯萌新..,请问一下大大，我安装了python3.12,也用powershell执行了pip install -r requirements.txt，用idle改了Main的账户密码，但是main自始自终都是打开来没反应orz...

**Reply by nightzora** [reply]
啊...问题解决了，是因为系统默认python不是3.12导致食用了过期发霉的python导致的（

**Comment by 7M7666** [comment]
![image](https://github.com/user-attachments/assets/b39fa1e8-9bc7-46ef-9dda-04cad69d9ea9)
解决ModuleNotFoundError: No module named 'urllib3.packages.six'
pip3 uninstall urllib3 -y --cert root.pem
pip3 install --no-cache-dir -U urllib3

**Comment by niceboy1we** [comment]
老哥，
![屏幕截图 2024-12-11 235900](https://github.com/user-attachments/assets/251d5387-07a8-486c-b7c7-5fe28a19c008)
这啥情况啊

**Reply by niceboy1we** [reply]
我纯小白对python和js可以说从没接触过，就高中微机课拿python做那些数学题过

**Reply by shuangjin1208** [reply]
这是没安装“requests”，你可以拿个ai软件问这种错误，差不多能解决70%的

**Reply by niceboy1we** [reply]
> 这是没安装“requests”，你可以拿个ai软件问这种错误，差不多能解决70%的

谢谢老哥

**Comment by wei-yy-cpu** [comment]
|##Captcha required
  这什么原因

**Reply by wei-yy-cpu** [reply]
已解决，原来是卡验证了

**Reply by 98fqg9m** [reply]
卡验证怎么解决

**Reply by LFklein** [reply]
自己进网页手动验证一下重新开始脚本就好

**Comment by CoGre105** [comment]
我点任何一个.py的文件为啥都是闪现一下然后消失了呢，要怎么正确打开

**Reply by usernameisnotavaliablee** [reply]
终端里运行

**Comment by 98fqg9m** [comment]
---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2025年4月22日(周二) 晚上10:53
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [VermiIIi0n/fuckZHS] 常见问题 (Discussion #25)




 
我点任何一个.py的文件为啥都是闪现一下然后消失了呢，要怎么正确打开

**Comment by shilezhi66** [comment]
|  |  |##Failed: code: -1 msg: 参数解码发生错误
这是什么原因

**Comment by jim-orange** [comment]
各位大佬好~

俺这儿刷着刷着401了 不知道啥情况...

<img width="745" height="389" alt="111" src="https://github.com/user-attachments/assets/79508c23-46b1-44d6-bdb2-01be51297c91" />
<img width="745" height="389" alt="222" src="https://github.com/user-attachments/assets/1053f270-4ee6-4ee5-ae9f-516426336d34" />

**Comment by Nur4388** [comment]
pycharm，main代码运行后二维码太大怎么解决，还有那个账号密码也要手动输入吗

**Reply by Melgiades** [reply]
在设置里找到ctrl➕滚轮可以缩放的选项打开
是的

**Comment by ExquisiteCore** [comment]
https://github.com/user-attachments/assets/34d43c15-f445-4fc3-8219-d0a636bbf544。忽然就这样了

**Comment by Norph1n** [comment]
随机暂停一分钟这条指令是哪句指令啊，我实在找不到，求大佬解答

**Comment by wangyihuangyang** [comment]
<img width="506" height="311" alt="Screenshot 2026-06-10 105836" src="https://github.com/user-attachments/assets/dd8da156-19b6-4f86-bb05-ab6a935e232a" />
这是啥原因啊，还没刷完就报错结束了

**Reply by usernameisnotavaliablee** [reply]
看起来像代码的bug？

**Reply by wangyihuangyang** [reply]
是bug,我用Claude改好了
![Uploading image.png…]()

**Comment by Criszzzz77** [comment]
新出的课程似乎-f检索不到？

**Reply by wangyihuangyang** [reply]
![Uploading image.png…]()
在这里

**Reply by wangyihuangyang** [reply]
好像没发出去，你打开智慧树的具体课程网址，网址链接像这种https://wisdom-mooc.zhihuishu.com/study/index?recruitAndCourseId=4e5f5858435c4859454a585958425d4350，那么id就是等号后面的4e5f5858435c4859454a585958425d4350这样
